File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/pojo/FillWordBuilder.java
package cz.muni.fi.umimecesky.pojo;

public class FillWordBuilder {
    private long id;
    private String wordMissing;
    private String wordFilled;
    private String variant1;
    private String variant2;
    private int correctVariant;
    private String explanation;
    private int grade;
    private boolean visibility;

    public FillWordBuilder id(long id) {
        this.id = id;
        return this;
    }

    public FillWordBuilder wordMissing(String wordMissing) {
        this.wordMissing = wordMissing;
        return this;
    }

    public FillWordBuilder wordFilled(String wordFilled) {
        this.wordFilled = wordFilled;
        return this;
    }

    public FillWordBuilder variant1(String variant1) {
        this.variant1 = variant1;
        return this;
    }

    public FillWordBuilder variant2(String variant2) {
        this.variant2 = variant2;
        return this;
    }

    public FillWordBuilder correctVariant(int correctVariant) {
        this.correctVariant = correctVariant;
        return this;
    }

    public FillWordBuilder explanation(String explanation) {
        this.explanation = explanation;
        return this;
    }

    public FillWordBuilder grade(int grade) {
        this.grade = grade;
        return this;
    }

    public FillWordBuilder visibility(boolean visibility) {
        this.visibility = visibility;
        return this;
    }

    public FillWord createFillWord() {
        return new FillWord(id, wordMissing, wordFilled, variant1, variant2, correctVariant, explanation, grade, visibility);
    }
}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/pojo/FillWordBuilder.kt
package cz.muni.fi.umimecesky.pojo

class FillWordBuilder {
    private var id: Long = 0
    private lateinit var wordMissing: String
    private lateinit var wordFilled: String
    private lateinit var variant1: String
    private lateinit var variant2: String
    private var correctVariant: Int = 0
    private lateinit var explanation: String
    private var grade: Int = 0
    private var visibility: Boolean = false

    fun id(id: Long): FillWordBuilder {
        this.id = id
        return this
    }

    fun wordMissing(wordMissing: String): FillWordBuilder {
        this.wordMissing = wordMissing
        return this
    }

    fun wordFilled(wordFilled: String): FillWordBuilder {
        this.wordFilled = wordFilled
        return this
    }

    fun variant1(variant1: String): FillWordBuilder {
        this.variant1 = variant1
        return this
    }

    fun variant2(variant2: String): FillWordBuilder {
        this.variant2 = variant2
        return this
    }

    fun correctVariant(correctVariant: Int): FillWordBuilder {
        this.correctVariant = correctVariant
        return this
    }

    fun explanation(explanation: String): FillWordBuilder {
        this.explanation = explanation
        return this
    }

    fun grade(grade: Int): FillWordBuilder {
        this.grade = grade
        return this
    }

    fun visibility(visibility: Boolean): FillWordBuilder {
        this.visibility = visibility
        return this
    }

    fun createFillWord(): FillWord =
            FillWord(id, wordMissing, wordFilled, variant1, variant2, correctVariant, explanation, grade, visibility)
}